java/ql/src/experimental/Security/CWE/CWE-918/RequestForgery.qll

```diff
@@ -167,6 +167,50 @@ private class RequestEntityUriArg extends RequestForgerySink {
   }
 }
 
+/**
+ * A class representing a sink for remote class loading mechanism.
+ * You can create `java.net.URLClassloader` instance either with new operator
+ * or with newInstance static method.
+ */
+private class URLClassloaderSink extends RequestForgerySink {
+  URLClassloaderSink() {
+    // Example:
+    // URLClassLoader urlClassLoader = new URLClassLoader(new URL[]{url})
+    exists(ClassInstanceExpr e |
```

Comment:
I think the `sink` here should not be the first parameter of `URLClassLoader`, because `new URLClassLoader(new URL[](url))` will not trigger a url request.

I think the `sink` here should be `ClassLoader.loadClass(String name)` or `URLClassLoader.findResource(String name)` etc. Because these calls will actually trigger url requests.

Reply:
So I can check if URL is controlled and loadClass/findResource method is called

Reply:
Yes

Reply:
@haby0 is right: constructing a class instance should not be the sink. It should propagate taint from the appropriate parameters to the class instance that it creates. You can do this fairly easily by making a class that extends `Constructor` and `TaintPreservingCallable`, as is done here: https://github.com/github/codeql/blob/8606386c2ce297221e08ff9cf81c261225a0d1c5/java/ql/src/semmle/code/java/dataflow/FlowSteps.qll#L75-L79 . The sink should then be methods like `loadClass` and `findResource`, as @haby0 said.